### PR TITLE
[IR-756] Genesis Block functionality for Kura.

### PR DESCRIPTION
**Pull request by humb1t**
Signed-off-by: Nikita Puzankov <humb1t@yandex.ru>

### CODE REVIEW

File: iroha/src/kura.rs [hunk 1]
```diff
@@ -37,6 +39,24 @@ impl Kura {
...
+        block_sender: CommittedBlockSender,
+        genesis_block: ValidBlock,
+    ) -> Self {
+        Kura {
+            mode,
+            block_store: BlockStore::with_genesis_block(block_store_path, genesis_block),
+            block_sender,
+            merkle_tree: MerkleTree::new(),
+            blocks: Vec::new(),
+        }
```
Comment (eadventurous) [inline iroha/src/kura.rs:57]: I think you forgot to notify `wsv` about the new block. We should send `genesis_block` through `block_sender` to `wsv`.

File: iroha/src/kura.rs [hunk 2]
```diff
@@ -162,6 +189,30 @@ mod tests {
...
+            .validate(&WorldStateView::new(Peer::new(
+                PeerId {
+                    address: "127.0.0.1:8080".to_string(),
+                    public_key: [0; 32],
+                },
+                &Vec::new(),
+            )))
+            .expect("Failed to validate block.");
+        assert!(
+            Kura::with_genesis_block(Mode::Strict, temp_dir.path(), tx, genesis_block)
```
Comment (eadventurous) [inline iroha/src/kura.rs:209]: Maybe also `assert` that block store contains this genesis block?

File: iroha/src/kura.rs (hunk 1 above)
Comment (humb1t) [reply iroha/src/kura.rs:57]: @eadventurous it will be done during `init`

**Pull request merged**